### PR TITLE
fix(ci/cd): Fix failing test packages for ARM CI runs

### PR DESCRIPTION
**Pull request by adityasadalage**
## Problem
We are currently skipping some test packages in ARM CI to get passing test runs. This prevents the ARM CI from having true parity with amd64 CI runs and we need a baseline for ARM tests as well as amd64.

## Solution
The fix for this is to update the lambda image to have an ARM binary as part of the "latest" tag. There is also an old dockerfile in e2e/custom_logic/cmd that needs to be updated to work with arm machines.

### CODE REVIEW

**Comment by skrdgraph** [comment]
we will have to push the arm64 lambda image out before merging this right?

**Comment by adityasadalage** [comment]
> 



> we will have to push the arm64 lambda image out before merging this right?

Yes thats correct

File: graphql/e2e/directives/docker-compose.yml
Comment (matthewmcneely) [inline graphql/e2e/directives/docker-compose.yml:40]: Any push would clobber the amd with the same tag. We need to use the `docker buildx` extensions. See this issue for some context: https://github.com/dgraph-io/dgraph-lambda/issues/36

**Review by matthewmcneely — APPROVED** [review]

File: graphql/e2e/directives/docker-compose.yml
Comment (adityasadalage) [reply graphql/e2e/directives/docker-compose.yml:40]: Yes I have done the buildx method on a forked repo according to https://github.com/dgraph-io/dgraph-lambda/issues/36. Just waiting on the go ahead to push the image for arm/amd platforms

**Review by joshua-goldstein — CHANGES_REQUESTED** [review]
There are places in the test suite where the lambda image is being pulled but not with the latest tag (I believe it uses the 1.40 tag or something similar).  These will have to be modified as well.

File: graphql/e2e/directives/docker-compose.yml
Comment (skrdgraph) [reply graphql/e2e/directives/docker-compose.yml:40]: Resolving this .. as it's done based on this https://github.com/dgraph-io/dgraph-lambda/issues/36

**Comment by skrdgraph** [comment]
> > There are places in the test suite where the lambda image is being pulled but not with the latest tag (I believe it uses the 1.40 tag or something similar). These will have to be modified as well. Also, please link this PR to a PR in the lambda repo that addresses the CD issues we discussed (which will have to be merged first).
> 
> Yeah, currently creating and testing lambda cd steps and will link it here

@adityasadalage we should merge this PR only when all the references to `1.4.0` is adjusted correctly in this PR itself.

**Comment by adityasadalage** [comment]
> > > There are places in the test suite where the lambda image is being pulled but not with the latest tag (I believe it uses the 1.40 tag or something similar). These will have to be modified as well. Also, please link this PR to a PR in the lambda repo that addresses the CD issues we discussed (which will have to be merged first).
> > 
> > 
> > Yeah, currently creating and testing lambda cd steps and will link it here
> 
> @adityasadalage we should merge this PR only when all the references to `1.4.0` is adjusted correctly in this PR itself.

Changed all references to dgraph-lambda to point to "latest"

**Review by joshua-goldstein — APPROVED** [review]
Previous failed CI runs were probably due to a lack of a rebase.  I rebased on the latest main:
```
git checkout main
git pull
git checkout adityasadalage/arm-ci-cd-changes-final
git rebase main
git push -f
```
and CI runs are now passing.

**Pull request merged**
